spring-cloud-gcp-core/src/main/java/org/springframework/cloud/gcp/core/util/MapBuilder.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2017-2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.core.util;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.springframework.util.Assert;
+
+/**
+ * Utility for building an immutable, type-safe {@link Map}.
+ * <p>Underlying implementation uses a {@link HashMap}, but disallows {@code null} keys,
+ * {@code null} values, and duplicate keys. This is consistent with Java 9 {@code Map.of()}
+ * static factory method, which will eventually replace the uses of this utility.
+ *
+ * @param <K> map key type
+ * @param <V> map value type
+ * @author Elena Felder
+ * @since 1.1
```

Comment:
New line before `@since`

Reply:
Do we separate all groups of identical tags? Codebase has different styles.

Reply:
What codebase?
I would like to have all the tag groups in the separate paragraph. Much easier to review and see what is going on.

The point is that in Open Source the code never stay the same. It is always evolves from version to version.
So, having it really readable and convenient for other people makes our project smoothly maintained.
And it becomes attractive for external contributions.